nav2_lifecycle_manager/src/lifecycle_manager_client.cpp

```diff
@@ -32,7 +33,9 @@ LifecycleManagerClient::LifecycleManagerClient(const std::string & name)
   active_service_name_ = name + std::string("/is_active");
 
   // Create the node to use for all of the service clients
-  node_ = std::make_shared<rclcpp::Node>(name + "_service_client");
+  std::string node_name = name + "_service_client";
+  std::replace(node_name.begin(), node_name.end(), '/', '_');
```

Comment:
Where is there a `/` in the node name used? I can't find any examples and leading with an `_` would make that private context potentially in the ROS2 naming style

Reply:
Nowhere in nav2, but I am using `nav2_lifecycle_manager::LifecycleManagerClient`s to start/pause/resume `nav2_util::LifecycleNodes`s which are under a namespace.
Like here:
https://github.com/ros-planning/navigation2/blob/d537cb186dd61a9491c0b1c7865f1e4e6891ca45/nav2_system_tests/src/updown/test_updown.cpp#L25-L31
but with the difference that `lifecycle_manager_navigation` and `lifecycle_manager_localization` would be found at `my_ns/lifecycle_manager_navigation` and `my_ns/lifecycle_manager_localization`.
And if currently, doing this: 
```
 int main(int argc, char ** argv) 
 { 
   rclcpp::init(argc, argv); 
   RCLCPP_INFO(rclcpp::get_logger("test_updown"), "Initializing test"); 
   nav2_lifecycle_manager::LifecycleManagerClient client_nav("my_ns/lifecycle_manager_navigation"); 
   nav2_lifecycle_manager::LifecycleManagerClient client_loc("my_ns/lifecycle_manager_localization");
```
would throw an error:

>  terminate called after throwing an instance of 'rclcpp::exceptions::InvalidNodeNameError'
>   what():  Invalid node name: node name must not contain characters other than alphanumerics or '_':
>   'my_ns/lifecycle_manager_navigation_service_client'

Reply:
So you want it to change to `my_ns_lifecycle_manager_navigation`? Why not just update the input string you're hardcoding into your main function? This doesn't seem like it would be good to mask from users if they  were genuinely expecting a namespaced output. 

https://docs.ros2.org/dashing/api/rclcpp/classrclcpp_1_1Node.html you can add a namespace to a field -- we could just have the lifecycle manager take in a potential namespace as an argument in addition to the name if you like (and just default to empty string)

Reply:
> So you want it to change to `my_ns_lifecycle_manager_navigation`? Why not just update the input string you're hardcoding into your main function? This doesn't seem like it would be good to mask from users if they were genuinely expecting a namespaced output.

Cause the constructor needs the full address of the lifecycle manager in order to initialize `manager_client_` and `is_active_client_` : 
https://github.com/ros-planning/navigation2/blob/d537cb186dd61a9491c0b1c7865f1e4e6891ca45/nav2_lifecycle_manager/src/lifecycle_manager_client.cpp#L29-L40

> https://docs.ros2.org/dashing/api/rclcpp/classrclcpp_1_1Node.html you can add a namespace to a field -- we could just have the lifecycle manager take in a potential namespace as an argument in addition to the name if you like (and just default to empty string)

That would indeed be a proper solution if you agree on changing the constructor. So it would become: `LifecycleManagerClient(const std::string & name, const std::string & namespace = "")` ?

Reply:
yeah, but if you're hiding `my_ms/topic` to `my_ns_topic`, then you **wouldn't** know the full address if you didn't know this masking was happening in the backend. Yes, we can change the interface.